test/hotspot/jtreg/compiler/loopopts/superword/TestMulAddS2I.java

```diff
@@ -75,15 +74,16 @@ public static void compare(int[] out) {
     }
 
     @Test
-    @IR(applyIfCPUFeature = {"sse2", "true"}, applyIf = {"UseUnalignedLoadStores", "true"},
+    @IR(applyIfCPUFeature = {"sse2", "true"},
+        applyIfPlatform = {"64-bit", "true"},
         counts = {IRNode.MUL_ADD_S2I, "> 0", IRNode.MUL_ADD_VS2VI, "> 0"})
-    @IR(applyIfCPUFeature = {"sse2", "true"}, applyIf = {"UseUnalignedLoadStores", "false"},
-        failOn = {IRNode.MUL_ADD_VS2VI}, // Can only pack LoadS if UseUnalignedLoadStores is true (default if sse4.2)
-        counts = {IRNode.MUL_ADD_S2I, "> 0"})
-    @IR(applyIfCPUFeature = {"asimd", "true"}, applyIf = {"MaxVectorSize", "16"}, // AD file requires vector_length = 16
-            counts = {IRNode.MUL_ADD_S2I, "> 0", IRNode.MUL_ADD_VS2VI, "> 0"})
-    @IR(applyIfCPUFeature = {"avx512_vnni", "true"}, applyIf = {"UseUnalignedLoadStores", "true"},
-            counts = {IRNode.MUL_ADD_S2I, "> 0", IRNode.MUL_ADD_VS2VI_VNNI, "> 0"})
+    @IR(applyIfCPUFeature = {"asimd", "true"},
+        applyIfPlatform = {"64-bit", "true"},
```

Comment:
Maybe we can drop the vm-bit check. The condition `(applyIfCPUFeature = {"asimd", "true"}` should hold only on `aarch64` platform.